modules/jaeger-thrift-http-exporter/src/main/scala/io/janstenpickle/trace4cats/jaeger/JaegerHttpSpanExporter.scala

```diff
@@ -0,0 +1,43 @@
+package io.janstenpickle.trace4cats.jaeger
+
+import cats.Foldable
+import cats.effect.kernel.Async
+import cats.syntax.either._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import io.jaegertracing.thriftjava.{Batch => JaegerBatch, Process, Span}
+import io.janstenpickle.trace4cats.`export`.HttpSpanExporter
+import io.janstenpickle.trace4cats.kernel.SpanExporter
+import io.janstenpickle.trace4cats.model.{Batch, TraceProcess}
+import org.apache.thrift.TSerializer
+import org.http4s.client.Client
+import org.http4s.headers.`Content-Type`
+import org.http4s.{Header, MediaType, Uri}
+
+object JaegerHttpSpanExporter {
+
+  private val headers: List[Header.ToRaw] = List(`Content-Type`(MediaType.application.`vnd.apache.thrift.binary`))
+
+  private val serializer = new TSerializer()
+
+  private def makePayload[G[_]: Foldable](jaegerProcess: Process, batch: Batch[G]): Array[Byte] = {
+    val spans = batch.spans.foldLeft(new java.util.ArrayList[Span]) { case (acc, span) =>
+      acc.add(JaegerSpan.convert(span))
+      acc
+    }
+    val jaegerBatch = new JaegerBatch(jaegerProcess, spans)
+    serializer.serialize(jaegerBatch)
```

Comment:
Can the call to `serialize` throw? If so I this will need handling with an `ApplicativeThrow` or `MonadThrow`. I think you should be able to swap `foldLeft` for `foldLeftM`